pkg/occlient/occlient.go

```diff
@@ -1682,23 +1685,37 @@ func (c *Client) WaitForBuildToFinish(buildName string) error {
 		return errors.Wrapf(err, "unable to watch build")
 	}
 	defer w.Stop()
+	timeout := time.After(OcBuildTimeout)
 	for {
```

Comment:
Don't understand what's going on in this `for` loop either. :disappointed:

Reply:
I can help here a bit. So here first thing to observe is the `select` statement on a quick TLDR: select statements are like channel collectors - listen on multiple channels at once, more here https://tour.golang.org/concurrency/5. 

now the `Build().Watch()` returns a channel which returns messages on as per the pod status. Now the change that has been done here is 
we wait for the pod to start before we start following the logs.

Reply:
as for the need for the `following`, The `w.ResultChan()` can send multiple messages of type `BuildPhaseRunning`. But we dont have to follow logs multiple times, just the first time should be good enough. Hence this flag.
Think of it as a singleton